libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -308,20 +327,24 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_WriteSync )
     lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetMasterConfig, &xOrigConfig);
     TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 
-    /* configure bus */
-    xTestConfig.ulFreq = SPI_TEST_FREQ;
-    xTestConfig.eMode = xtestDefaultConfigMode;
-    xTestConfig.eSetBitOrder = xtestDefaultconfigBitOrder;
-    xTestConfig.ucDummyValue = 0;
-    lRetVal = iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xTestConfig);
-    TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+    if( TEST_PROTECT() )
+    {
+        /* configure bus */
+        xTestConfig.ulFreq = ultestIotSPIFrequency;
+        xTestConfig.eMode = xtestIotSPIDefaultConfigMode;
+        xTestConfig.eSetBitOrder = xtestIotSPIDefaultconfigBitOrder;
+        xTestConfig.ucDummyValue = ultestIotSPIDummyValue;
 
-    lRetVal = iot_spi_write_sync( xSPIHandle, ucTxBuf, 4);
-    TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        lRetVal = iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xTestConfig);
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 
-    //lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetTxNoOfbytes, &xBytesTx);
-    //TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
-    //TEST_ASSERT_EQUAL( xBytesTx, 4 );
+        lRetVal = iot_spi_write_sync( xSPIHandle, ucTxBuf, 4);
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+
+        lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetTxNoOfbytes, &xBytesTx);
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        TEST_ASSERT_EQUAL( xBytesTx, 4 );
```

Comment:
Same here. for sync APIs, TxNoOfBytes may not return number of bytes transmitted. Its used for async calls. Sync API is supposed to return SUCCESS only if all the bytes are sent.

Reply:
Why can't it be used with sync? What if a partial amount of the operation was sent and we wanted to see how much of our data was able send?